src/CSV.jl

```diff
@@ -425,6 +428,31 @@ function file(source,
     else
         typecodes = TypeCode[T for _ = 1:ncols]
     end
+    # set any unselected columns to typecode USER | MISSING
+    todrop = Int[]
+    if select !== nothing
+        if select isa AbstractVector{Int}
+            foreach(1:ncols) do i
+                i in select || push!(todrop, i)
+            end
+        elseif select isa AbstractVector{Symbol} || select isa AbstractVector{<:AbstractString}
+            select = map(Symbol, select)
+            foreach(1:ncols) do i
+                ix = findfirst(==(names[i]), select)
+                ix === nothing && push!(todrop, i)
+            end
+        elseif select isa Base.Callable
+            foreach(1:ncols) do i
+                select(i, names[i]) || push!(todrop, i)
+            end
+        else
+            error("`select` keyword argument must be an `AbstractVector` of `Int`, `Symbol`, or `String`, or a selector function of the form `(i, name) -> keep::Bool`")
+        end
+    end
+    sort!(unique!(todrop))
```

Comment:
This shouldn't be needed since `push!` is only called once for each `i` and in the right order.